app/controllers/responsiblepeople/PositionWithinBusinessStartDateController.scala

```diff
@@ -0,0 +1,124 @@
+                  => Ok(position_within_business_start_date(Form2[PositionStartDate](startDate), edit, index, bt,  personName.titleName, positions, displayNominatedOfficer(rp, hasNominatedOfficer(data)), flow))
+                case Some(rp@ResponsiblePerson(Some(personName),_,_,_,_,_,_,_,_,_, Some(Positions(positions, None)),_,_,_,_,_,_,_,_,_,_,_))
+                  => Ok(position_within_business_start_date(EmptyForm, edit, index, bt, personName.titleName, positions, displayNominatedOfficer(rp, hasNominatedOfficer(data)), flow))
+                case _
+                  => NotFound(notFoundView)
+              }
+            }).getOrElse(NotFound(notFoundView))
+          }
+    }
+
+  def post(index: Int, edit: Boolean = false, flow: Option[String] = None) = Authorised.async {
+    implicit authContext =>
+      implicit request =>
+        Form2[PositionStartDate](request.body) match {
+          case f: InvalidForm =>
+            dataCacheConnector.fetchAll map { optionalCache =>
+              (optionalCache map { cache =>
+
+                val bt = ControllerHelper.getBusinessType(cache.getEntry[BusinessMatching](BusinessMatching.key))
+                  .getOrElse(BusinessType.SoleProprietor)
+
+                val data = cache.getEntry[Seq[ResponsiblePerson]](ResponsiblePerson.key)
+
+                getResponsiblePersonFromData(data,index) match {
+                  case s@Some(rp@ResponsiblePerson(Some(personName),_,_,_,_,_,_,_,_,_, Some(Positions(positions,_)),_,_,_,_,_,_,_,_,_,_,_)) =>
+                    BadRequest(position_within_business_start_date(f, edit, index, bt, ControllerHelper.rpTitleName(s), positions, displayNominatedOfficer(rp, hasNominatedOfficer(data)), flow))
+                  case _
+                    => NotFound(notFoundView)
+                }
+              }).getOrElse(NotFound(notFoundView))
+            }
+          case ValidForm(_, data) => {
+            for {
+              _ <- updateDataStrict[ResponsiblePerson](index) { rp => rp.positions match {
+                  case Some(x) => rp.positions(Positions.update(x, data))
+                  case None => throw new IllegalStateException("Positions does not exist, cannot update start date")
+                }
+              }
+              rpSeqOption <- dataCacheConnector.fetch[Seq[ResponsiblePerson]](ResponsiblePerson.key)
+            } yield {
+                edit match {
+                  case true => Redirect(routes.DetailedAnswersController.get(index, flow))
+                  case _ => Redirect(routes.SoleProprietorOfAnotherBusinessController.get(index, edit, flow))
+                }
+            }
+          } recoverWith {
+            case _: IndexOutOfBoundsException => Future.successful(NotFound(notFoundView))
+          }
+        }
+  }
+
+  private[controllers] def hasNominatedOfficer(rpSeqOption: Option[Seq[ResponsiblePerson]]): Boolean = {
```

Comment:
this might not need to change here and now. but IMHO, this is feature envy and could/should be part of the ResponsiblePeople object.

Reply:
I've removed these methods from both Controllers and added to the ResponsiblePerson object.